### ISSUE TITLE
Display compatibility with hs-minor-mode

### ISSUE THREAD

**Issue by dabrahams** [alpaker/fill-column-indicator#72]
If you customize `hs-set-up-overlay` as suggested in its documentation

```lisp
(defun dwa/hs-ellipsis (ov)
  (when (overlay-get ov 'hs)
    (overlay-put ov 'display "▷…")))

(setq hs-set-up-overlay 'dwa/hs-ellipsis)
```

and then hide some things that cross line breaks, you get this effect:

<img width="769" alt="fci" src="https://cloud.githubusercontent.com/assets/44065/20235552/078047e2-a84a-11e6-8606-164343186da0.png">

when without `fci-mode`, it looks like this:

<img width="661" alt="no-fci" src="https://cloud.githubusercontent.com/assets/44065/20235579/9eb456e4-a84a-11e6-8be8-dfe137a3a1cc.png">

My theory, not having looked at the code, is that fci-mode needs to skip over hidden newlines when determining where to place the indicators.

This could be related to #54

**Comment by alpaker** [comment]
Hi Dave.  Thanks for the report.  I'm about 85% done with a rewrite of the package and will take a look at this issue after that's complete, but to be entirely honest compatibility might be hard to achieve.  Conflict with other modes that alter end-of-line display is getting more and more and difficult to avoid.  That's ultimately not surprising, as `fci-mode` is trying to create something like a background canvas on which to render buffer text, but does so using a mechanism--overlays--that's fundamentally an inline display mechanism.  The impedance mismatch is such that Eli Zarestskii has declared that he hates working on display engine bugs associated with this mode.

All of which is to say that I'm considering declaring `fci-mode` to be end of life and turning to a reimplementation of the mode's functionality that hooks in to the display engine in a different way.

Sorry to veer off into an existential discussion of the future of the package.  I just want to make sure expectations are realistic.

**Comment by alpaker** [comment]
Just FYI, I'm abandoning this package and welcoming requests to be take over as maintainer.

**Comment by alpaker** [comment]
The functionality provided by this package has now been implemented natively in Emacs as `display-fill-column-indicator-mode`, available as of version 27.0.90. Unless you are forced to use an older Emacs, you should use the native implementation instead of `fci-mode`. The former is unaffected by most of the compatibility issues that this add-on has.
